be/src/vec/exec/join/join_op.h

```diff
@@ -22,19 +22,42 @@
 #include "vec/core/block.h"
 
 namespace doris::vectorized {
+
 /// Reference to the row in block.
 struct RowRef {
     using SizeT = uint32_t; /// Do not use size_t cause of memory economy
 
-    const Block* block = nullptr;
-    SizeT row_num = 0;
-    // Use in right join to mark row is visited
-    // TODO: opt the varaible to use it only need
-    bool visited = false;
+    // using union to represent two cases
+    // 1. when RowRefList containing only one RowRef, visited + blockptr are valid
+    // 2. when RowRefList contaning multi RowRef, it's used through next pointer
+    union { 
+        struct {
+            uint64_t visited : 1;
```

Comment:
Why is it defined as uint64 here, not use automatic memory alignment?

Reply:
i need 64bit wide to union with void*